docs/guide/uploading-downloading-binary-content.md

```diff
@@ -0,0 +1,147 @@
+# Uploading & Downloading Binary Content | AngularFire Guide
+
+## Table of Contents
+
+* [Overview](#overview)
+* [API Summary](#api-summary)
+* [Uploading Files](#uploading-files)
+* [Displaying Images with the `firebase-src` Directive](#displaying-images-with-the-firebase-src-directive)
+* [Retrieving Files from the Template](#retrieving-files-from-the-template)
+
+## Overview
+
+Firebase provides [a hosted binary storage service](https://firebase.google.com/docs/storage/)
+which enables you to store and retrieve user-generated content like images, audio, and
+video directly from the Firebase client SDK.
+
+Binary files are stored in a Firebase Storage bucket, not in the Realtime Database.
+The files in your bucket are stored in a hierarchical structure, just like
+in the Realtime Database.
+
+To use the Firebase Storage binding, first [create a Firebase Storage reference](https://firebase.google.com/docs/storage/web/create-reference).
+Then, using this reference, pass it into the `$firebaseStorage` service:
+
+```js
+// define our app and dependencies (remember to include firebase!)
+angular
+  .module("sampleApp", [
+    "firebase"
+  ])
+  .controller("SampleCtrl", SampleCtrl);
+
+// inject $firebaseStorage into our controller
+function SampleCtrl($firebaseStorage) {
+  // create a Firebase Storage Reference for the $firebaseStorage binding
+  var storageRef = firebase.storage().ref("userProfiles/physicsmarie");
+  var storage = $firebaseStorage(storageRef);
+}
+SampleCtrl.$inject = ["$firebaseStorage"];
+```
+
+## API Summary
+
+The Firebase Storage service is created with several special `$` methods, all of which are listed in the following table:
+
+| Method  | Description |
+| ------------- | ------------- |
+| [`$put(file, metadata)`](/docs/reference.md#putfile-metadata) |	Uploads file to configured path with optional metadata. Returns an AngularFire wrapped `UploadTask`. |
```

Comment:
Let's link `UploadTask` to `#upload-task-api-summary` everywhere in this API summary (four instances).